### PR TITLE
Load batch by batch

### LINKED ISSUE #25142: [Enhancement]: Load data batch by batch
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What would you like to be added?

Now Milvus loads all data once and copies the data into segment, which takes about x2 peak memory usage.

This makes the mmap optimization useless as it still takes x1 memory usage even with mmap enabled, I'm going to change the load logic, to load data batch by batch, like the ArrayBuilder of Arrow

### Why is this needed?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by yah01**
This will significantly reduce the memory usage while loading
- 1x memory usage and MBs overhead for buffer (memory mode)
- only MBs overhead for buffer (mmap mode)

resolve #25142

### CODE REVIEW

**Comment by yah01** [comment]
rerun-ut

**Comment by yah01** [comment]
rerun ut

**Comment by yah01** [comment]
rerun ut

File: internal/core/src/common/Channel.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    void
+    push(const T& value) {
+        inner_.push(value);
+    }
+
+    bool
+    pop(T& value) {
+        thread_local std::optional<T> result;
```
Comment (jiaoew1991) [inline internal/core/src/common/Channel.h]: Are there any other methods to replace threadlocal implementation?

File: internal/querynodev2/segments/segment_loader.go [hunk 2]
```diff
@@ -243,18 +244,21 @@ func (loader *segmentLoader) Load(ctx context.Context,
 			return nil, errors.New("segment was removed from the loading map early")
 		}
 
+		log.Info("wait segment loaded...", zap.Int64("segmentID", segment.GetSegmentID()))
```
Comment (jiaoew1991) [inline internal/querynodev2/segments/segment_loader.go:247]: Will there be too much info in the logs?

Comment (yah01) [reply internal/querynodev2/segments/segment_loader.go:247]: won't, as the log is for load only, and there are few segments

**Pull request merged**